### PR TITLE
Github Actions 1: Cross Platform Build 

### PR DESCRIPTION
**Pull request by nbollis**
Created a github action that builds MetaMorpheus on Windows, Linux, and mac. 

To build on windows and mac, a new configuration was generated that excluded the MetaMorphesusSetup and Project.wix Projects

### CODE REVIEW

File: MetaMorpheus/GUI/GUI.csproj [hunk 1]
```diff
@@ -12,16 +12,21 @@
     <GenerateAssemblyFileVersionAttribute>false</GenerateAssemblyFileVersionAttribute>
     <RootNamespace>MetaMorpheusGUI</RootNamespace>
     <AssemblyName>MetaMorpheusGUI</AssemblyName>
-    <Configurations>Debug;Release</Configurations>
+    <Configurations>Debug;Release;UbuntuMac</Configurations>
```
Comment (Alexander-Sol) [inline MetaMorpheus/GUI/GUI.csproj:15]: Surely the GUI won't work on linux/unix

File: MetaMorpheus/MetaMorpheusSetup/MetaMorpheusSetup.wixproj [hunk 2]
```diff
@@ -14,6 +14,12 @@
     <SuppressIces>
     </SuppressIces>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='UbuntuMac|x64'">
+    <PlatformTarget>x64</PlatformTarget>
```
Comment (Alexander-Sol) [inline MetaMorpheus/MetaMorpheusSetup/MetaMorpheusSetup.wixproj:18]: Wix builds .msi, which are specific to windows

File: MetaMorpheus/Test/SpectralRecoveryTest.cs [hunk 3]
```diff
@@ -377,7 +379,7 @@ public static void SpectralWriterTest()
             postSearchTask.Run();
 
             var libraryList = Directory.GetFiles(path, "*.*", SearchOption.AllDirectories);
-            string updateLibraryPath = libraryList.First(p => p.Contains("SpectralLibrary") && !p.Contains(matchingvalue)).ToString();
+            string updateLibraryPath = libraryList.First(p => p.Contains("updateSpectralLibrary") && !p.Contains(matchingvalue)).ToString();
```
Comment (Alexander-Sol) [inline MetaMorpheus/Test/SpectralRecoveryTest.cs:382]: Great change!

File: MetaMorpheus/MetaMorpheusSetup/MetaMorpheusSetup.wixproj (hunk 2 above)
Comment (nbollis) [reply MetaMorpheus/MetaMorpheusSetup/MetaMorpheusSetup.wixproj:18]: It needs to be defined in the csproj file, but the sln file tells it not to build wix and metamorpheus setup when running on UbuntuMac configuration

Comment (nbollis) [reply MetaMorpheus/MetaMorpheusSetup/MetaMorpheusSetup.wixproj]: removed this

File: MetaMorpheus/GUI/GUI.csproj (hunk 1 above)
Comment (nbollis) [reply MetaMorpheus/GUI/GUI.csproj]: removed this

**Review by Alexander-Sol — APPROVED** [review]

**Review by trishorts — APPROVED** [review]

**Review by elaboy — APPROVED** [review]

**Comment by acesnik** [comment]
Related issue for reference, https://github.com/smith-chem-wisc/MetaMorpheus/issues/2088

**Pull request merged**